rules/ruleset.go

```diff
@@ -50,6 +52,7 @@ var (
 	errRuleSetNotTombstoned = errors.New("ruleset is not tombstoned")
 	errRuleNotFound         = errors.New("rule not found")
 	errNoRuleSnapshots      = errors.New("rule has no snapshots")
+	errUnknownOpType        = merrors.NewInvalidInputError("changes must contain an op")
```

Comment:
I think we want to include the op type in the error message, granted that the op type may be empty if unspecified, but can be very useful when it is.